src/test/credentials/sso/model.test.ts

```diff
@@ -0,0 +1,73 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as assert from 'assert'
+import * as sinon from 'sinon'
+import * as vscode from 'vscode'
+import { createTestWindow } from '../../shared/vscode/window'
+import { openSsoPortalLink } from '../../../credentials/sso/model'
+import { assertTelemetry } from '../../testUtil'
+import { CancellationError } from '../../../shared/utilities/timeoutUtils'
+
+describe('openSsoPortalLink', function () {
+    let testWindow: ReturnType<typeof createTestWindow>
+    let openExternal: sinon.SinonStub<any[], Thenable<boolean>>
+
+    beforeEach(function () {
+        sinon.replace(vscode, 'window', (testWindow = createTestWindow()))
+        openExternal = sinon.stub(vscode.env, 'openExternal').resolves(true)
```

Comment:
We can probably de-dupe this into a test util.

Reply:
This is now done globally in https://github.com/aws/aws-toolkit-vscode/pull/3148/commits/e2d73035e521f290aa9af56abc391448cc369b12